File: csharp/剑指22链表中的第k个节点.cs

public partial class Solution
{
    public ListNode GetKthFromEnd(ListNode head, int k)
    {
        ListNode p = head;

        for (int i = 0; i < k; i++, p = p.next) ;

        for (; p!=null;head=head.next, p = p.next) ;
        return head;
    }
}

File: python/main/剑指22链表中的第k个节点.py
from linklist import ListNode

def getKthFromEnd(head: ListNode, k: int) -> ListNode:
    p=head

    for _ in range(k):
        p=p.next
    
    while(p!=None):
        p=p.next
        head=head.next
    
    return head
